controllers/ec2costoptimizer_controller.go

```diff
@@ -26,7 +35,9 @@ type Ec2CostOptimizerReconciler struct {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *Ec2CostOptimizerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	pred := predicate.GenerationChangedPredicate{}
 	return ctrl.NewControllerManagedBy(mgr).
+		WithEventFilter(pred).
```

Comment:
This predicate is to avoid triggering reconcile method on a status update. otherwise, it will enter into an infinite loop, where on each status update `Reconcile` will be called.

Reply:
Okay Great